.github/workflows/push_release.yml

```diff
@@ -0,0 +1,87 @@
+name: Main branch to Release Branch Sync PR
+
+on:
+  workflow_dispatch:
+    inputs:
+      org:
+        description: 'Select Organization'
+        required: true
+        type: choice
+        options:
+          - 'opendatahub.io'
+          - 'red-hat-data-services'
+      sync_repo:
+        description: 'Select Repository'
+        required: true
+        type: choice
+        options:
+          - 'modelmesh'
+          - 'caikit-tgis-serving'
+          - 'openvino'
+          - 'vllm'
+          - 'caikit-nlp'
+          - 'caikit'
+          - 'odh-model-controller'
+          - 'caikit-tgis-backend'
+          - 'caikit-nlp-client'
+          - 'model-registry'
+      main_branch:
+        description: 'Main branch to pull from'
+        default: 'main'
+        required: true
+      release_branch:
+        description: 'Branch to push to [release branch]'
+        required: true
+
+env:
+  ACTIONS_PAT: ${{ secrets.ACTIONS_PAT }}
+
+permissions:
+  # Need `contents: read` to checkout the repository
+  # Need `contents: write` to merge branches
+  contents: write
+  
+
+jobs:
+  push-release-branch:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Set up Git
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+            repository: ${{ github.event.inputs.org }}/${{ github.event.inputs.sync_repo }}
+            token: ${{ secrets.ACTIONS_PAT }}
+            fetch-depth: 0
+
+      
+      - name: Print current repository
+        run: git remote -v
+
+      - name: Fetch all branches
+        run: git fetch --all
+
+      - name: Check out release branch
+        run: |
+          git checkout ${{ github.event.inputs.release_branch }}
+          git fetch origin
+
+      - name: Merge main into release
```

Comment:
Again this part doesnt work . It causes the same problem as explained in the last comment. 

The issue is here once you try to open a PR it failes there as it is already synced in this step.

Reply:
but if without merge, it will only push the current state of the release branch without incorporating any changes from the main branch

Reply:
Removed Merge step